### PR TITLE
Tutorial to run NetBox with one (three) commands.

### PR DESCRIPTION
**Pull request by javierdelapuente**
### Overview

This PR creates a new tutorial, that includes a cloud init file that can be run with Multipass. This tutorial is an experiment for the simplest way to run NetBox.

To test the tutorial before this PR is merged to main, launch the VM with this command instead:
`# multipass launch  -vvvv --cloud-init https://raw.githubusercontent.com/canonical/netbox/multipass-info/charm/cloudinit-juju-3.1.yaml --timeout 1800 --name netbox --memory 4G --cpus 3 --disk 30G 22.04`

### Rationale

Make everything as simple as possible so users can test and experiment NetBox in the Juju ecosystem as fast as possible, and with as little context as possible.

The only requirement of this new tutorial is Multipass.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Manging Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: charm/docs/tutorial/test-netbox.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+Multipass can be easily installed in Linux, Windows and macOS. See
+https://multipass.run/install for more instructions.
+
+## Launch NetBox with a Multipass instance.
+
+
+Launch a VM named `netbox` with Multipass, using the cloud init configuration that will installed NetBox. This
+process will take about 20 minutes, depending on your computer and internet connection. Run the next command
+to install NetBox:
```
Comment (mthaddon) [inline charm/docs/tutorial/test-netbox.md:23]: It may be obvious, but we might want to list the requirements here (that you can allocate at least 4G RAM, 3 CPUs and 30G of disk to this VM). Or possibly there should be a "requirements" section earlier on in the document.

File: charm/docs/tutorial/test-netbox.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Testing NetBox
```
Comment (amandahla) [inline charm/docs/tutorial/test-netbox.md:1]: I couldn't find it but I think that there is a standard for writing tutorials starting with "What you'll do" like this:
https://github.com/canonical/jenkins-k8s-operator/blob/main/docs/tutorial/getting-started.md
Also, minor: file name should be equal to the title

File: charm/docs/tutorial/test-netbox.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+VMs with NetBox installed.
+
+
+Multipass can be easily installed in Linux, Windows and macOS. See
+https://multipass.run/install for more instructions.
+
+## Launch NetBox with a Multipass instance.
+
+
+Launch a VM named `netbox` with Multipass, using the cloud init configuration that will installed NetBox. This
```
Comment (amandahla) [inline charm/docs/tutorial/test-netbox.md:21]: Could the cloud init configuration be a link to the file?

File: charm/docs/tutorial/test-netbox.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+```
+multipass exec netbox -- juju run traefik-k8s/0 show-proxied-endpoints --format=yaml
+```
+
+You can create an admin user with the next command:
+```
+multipass exec netbox -- juju run netbox/leader create-superuser username=admin email=netbox@example.com
+```
+
+Once you are finished, you can delete the netbox VM with the next command:
```
Comment (amandahla) [inline charm/docs/tutorial/test-netbox.md:39]: Would make sense to add some example of action to test Netbox once the user is created?

File: charm/cloudinit-juju-3.1.yaml [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+  - snap install microk8s --channel 1.29-strict/stable
+  - snap alias microk8s.kubectl kubectl
+  - snap alias microk8s.kubectl k
+  - snap install yq
+  - snap install multipass-sshfs
+
+runcmd:
+
+- |
+  # disable swap
```
Comment (amandahla) [inline charm/cloudinit-juju-3.1.yaml:46]: Out of curiosity: Why is this needed?

File: charm/docs/tutorial/test-netbox.md (hunk 2 above)
Comment (erinecon) [reply charm/docs/tutorial/test-netbox.md]: I also couldn't find it, but I think this is correct. It's good practice to have a "What you'll do" section and a prerequisites section for tutorials

Comment (javierdelapuente) [reply charm/docs/tutorial/test-netbox.md]: similar in here: https://diataxis.fr/tutorials/#the-language-of-tutorials

File: charm/cloudinit-juju-3.1.yaml (hunk 5 above)
Comment (javierdelapuente) [reply charm/cloudinit-juju-3.1.yaml:46]: That's is basically a remnant from the charm-dev blueprint. In any case, it is probably a good practice, as I'd rather have process crashing that using swap memory...

File: charm/docs/tutorial/test-netbox.md (hunk 2 above)
Comment (javierdelapuente) [reply charm/docs/tutorial/test-netbox.md]: Improved the structure a bit!

File: charm/docs/tutorial/test-netbox.md (hunk 3 above)
Comment (javierdelapuente) [reply charm/docs/tutorial/test-netbox.md]: Done!

File: charm/docs/tutorial/test-netbox.md (hunk 4 above)
Comment (javierdelapuente) [reply charm/docs/tutorial/test-netbox.md]: I have added a few more text, not really sure what to write... Any other suggestion?

Comment (amandahla) [reply charm/docs/tutorial/test-netbox.md]: I'm not so familiar with netbox so while reading the tutorial I had the impression that you create the admin user and then you can delete the netbox VM. I missed something like "now you can login and then check that this is working using this and that command". What do you think?

File: charm/cloudinit-juju-3.1.yaml (hunk 5 above)
Comment (amandahla) [reply charm/cloudinit-juju-3.1.yaml:46]: Interesting. I ended up digging into the question and found this issue https://github.com/kubernetes/kubernetes/issues/53533

File: charm/docs/tutorial/test-netbox.md (hunk 4 above)
Comment (amandahla) [reply charm/docs/tutorial/test-netbox.md]: I wrote a comment before seeing the changes, now it's nice!

**Review by amandahla — APPROVED** [review]
LGTM

**Review by erinecon — APPROVED** [review]
Looks good
